### PR TITLE
Initial implementation of grader utils

### PR DESCRIPTION
**Pull request by bhuvy2**
Ideally we want to be able to keep it loosely coupled and install this in a virtualenv

```
import grader_utils
grader_utils.ProcessWrapper ...
```

Thoughts? Anything missing?

### CODE REVIEW

File: grader_utils/__init__.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        """Run the command as a separate process and return immediately.
+
+        Call join() later to get result of process. Timeout does not start
+        until join() is called. Additionally send the delayed_inputs and inputs
+        to process.
+        """
+        def target():
+            # Pass these inputs to STDIN with delays
+            for i in self.delayed_inputs:
+                if type(i) is int or type(i) is float:
```
Comment (aneeshdurg) [inline grader_utils/__init__.py:103]: I don't like this. Can we instead define a type like `Sleep` or something that stores a float and any other type gets cast to bytes or is rejected?

File: grader_utils/__init__.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        Will timeout based on self.timeout. Timeout involves sending SIGTERM
+        followed by SIGKILL if that is ignored.
+        """
+        self.thread.join(timeout=self.timeout)
+        alive = self.is_alive if self.check_pg_alive else self.thread.is_alive
+        self.exit_timeout = self.thread.is_alive()
+        if alive():
+            self.exit_timeout_pg = self.check_pg_alive
+            lg.info("{}: is still alive. Terminating now...".format(self.process.pid))
+            os.killpg(self.process.pid, signal.SIGTERM)
```
Comment (aneeshdurg) [inline grader_utils/__init__.py:141]: This can't be done with the Popen object?

**Review by aneeshdurg — CHANGES_REQUESTED** [review]
Couple minor things. There's still some features that I think could be added (like setting up magic env stuff)

**Comment by bhuvy2** [comment]
Ideally we don't add the magic env stuff here bc public. Can do everything else.

File: grader_utils/__init__.py (hunk 2 above)
Comment (bhuvy2) [reply grader_utils/__init__.py:141]: Apparently not

**Pull request merged**